infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -83,6 +101,20 @@ public KubernetesNamespace create(String workspaceId) throws InfrastructureExcep
     return namespace;
   }
 
+  protected String evalNamespaceName(String workspaceId, Subject currentUser) {
+    if (isNullOrEmpty(this.namespaceName)) {
+      return workspaceId;
+    } else {
+      String tmpNamespaceName = this.namespaceName;
+      for (String placeholder : NAMESPACE_NAME_PLACEHOLDERS.keySet()) {
```

Comment:
It may be a really little performance improvement but maybe it also could make code clearer as well if we check `isPredefined` before iterating placeholders?
I mean 
```java
protected String evalNamespaceName(String workspaceId, Subject currentUser) {
    if (isPredefined) {
      return this.namespaceName;
    } else if (isNullOrEmpty(this.namespaceName)) {
      return workspaceId;
    } else {
      String tmpNamespaceName = this.namespaceName;
      for (String placeholder : NAMESPACE_NAME_PLACEHOLDERS.keySet()) {
 tmpNamespaceName =
            tmpNamespaceName.replaceAll(
                placeholder, NAMESPACE_NAME_PLACEHOLDERS.get(placeholder).apply(currentUser));
      }
      return tmpNamespaceName;
    }
  }
```

Reply:
yes, very valid point. fixed.